MIGRATING.md

```diff
@@ -0,0 +1,242 @@
+Index keys consistency [\#568](https://github.com/CosmWasm/cw-plus/pull/568)
+
+For both `UniqueIndex` and `MultiIndex`, returned keys (deserialized or not) are now the associated *primary keys* of
+the data. This is a breaking change for `MultiIndex`, as the previous version returned a composite of the index key and
+the primary key.
+
+Examples of required changes are:
+```diff
+diff --git a/packages/storage-plus/src/indexed_map.rs b/packages/storage-plus/src/indexed_map.rs
+index 9f7178af..d11d501e 100644
+--- a/packages/storage-plus/src/indexed_map.rs
++++ b/packages/storage-plus/src/indexed_map.rs
+@@ -722,7 +722,7 @@ mod test {
+             last_name: "".to_string(),
+             age: 42,
+         };
+-        let pk1: &[u8] = b"5627";
++        let pk1: &str = "5627";
+         map.save(&mut store, pk1, &data1).unwrap();
+
+         let data2 = Data {
+@@ -730,7 +730,7 @@ mod test {
+             last_name: "Perez".to_string(),
+             age: 13,
+         };
+-        let pk2: &[u8] = b"5628";
++        let pk2: &str = "5628";
+         map.save(&mut store, pk2, &data2).unwrap();
+
+         let data3 = Data {
+@@ -738,7 +738,7 @@ mod test {
+             last_name: "Young".to_string(),
+             age: 24,
+         };
+-        let pk3: &[u8] = b"5629";
++        let pk3: &str = "5629";
+         map.save(&mut store, pk3, &data3).unwrap();
+
+         let data4 = Data {
+@@ -746,7 +746,7 @@ mod test {
+             last_name: "Bemberg".to_string(),
+             age: 43,
+         };
+-        let pk4: &[u8] = b"5630";
++        let pk4: &str = "5630";
+         map.save(&mut store, pk4, &data4).unwrap();
+
+         let marias: Vec<_> = map
+@@ -760,8 +760,8 @@ mod test {
+         assert_eq!(2, count);
+
+         // Remaining part (age) of the index keys, plus pks (bytes) (sorted by age descending)
+-        assert_eq!((42, pk1.to_vec()), marias[0].0);
+-        assert_eq!((24, pk3.to_vec()), marias[1].0);
++        assert_eq!(pk1, marias[0].0);
++        assert_eq!(pk3, marias[1].0);
+
+         // Data
+         assert_eq!(data1, marias[0].1);
+```
+
+- Remove the primary key from the `MultiIndex` key specification [\#533](https://github.com/CosmWasm/cw-plus/issues/533) /
+`MultiIndex` primary key spec removal [\#569](https://github.com/CosmWasm/cw-plus/pull/569)
+
+The primary key is no longer needed when specifying `MultiIndex` keys. This simplifies both `MultiIndex` definitions
+and usage.
+
+Required changes are along the lines of:
+```diff
+diff --git a/packages/storage-plus/src/indexed_map.rs b/packages/storage-plus/src/indexed_map.rs
+index 022a4504..c7a3bb9d 100644
+--- a/packages/storage-plus/src/indexed_map.rs
++++ b/packages/storage-plus/src/indexed_map.rs
+@@ -295,8 +295,8 @@ mod test {
+     }
+
+     struct DataIndexes<'a> {
+-        // Second arg is for storing pk
+-        pub name: MultiIndex<'a, (String, String), Data, String>,
++        // Last args are for signaling pk deserialization
```

Comment:
If I remember my comment well: https://github.com/CosmWasm/cw-plus/pull/569#discussion_r763850654
this should be a `type parameter` instead of `args`. :)

Reply:
Good eye. Took it from an old diff.